spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```
@@ -197,4 +198,87 @@ static Schema getOutputSchema(Schema inputSchema, String predictionField) {
+   */
+  static void validateFeatureGeneratorConfig(Schema inputSchema, String outputColumnMapping) {
+    Map<String, String> map;
+    try {
+      map = Splitter.on(',').trimResults().withKeyValueSeparator(":").split(outputColumnMapping);
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException(
+        String.format("Invalid output feature mapping. %s. Please make sure it is in the format " +
+                        "'input-column':'transformed-output-column'.", e.getMessage()), e);
+    }
+    for (Map.Entry<String, String> entry : map.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+      if (inputSchema.getField(key) == null) {
+        throw new IllegalArgumentException(String.format("Input field %s does not exist in the input schema %s.",
+                                                         key, inputSchema.toString()));
+      }
+      Schema schema = inputSchema.getField(key).getSchema();
+      Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+      Schema.Type componentSchemaType = null;
+      if (type.equals(Schema.Type.ARRAY)) {
+        Schema componentSchema = schema.getComponentSchema();
+        componentSchemaType = componentSchema.isNullable() ? componentSchema.getNonNullable().getType() :
+          componentSchema.getType();
+      }
+      if (!(type.equals(Schema.Type.STRING) || (type.equals(Schema.Type.ARRAY) &&
+        componentSchemaType.equals(Schema.Type.STRING)))) {
+        throw new IllegalArgumentException(String.format("Field to be transformed should be of type String or " +
+                                                           "Nullable String or Array of type String or Nullable " +
+                                                           "String . But was %s for field %s.", type, key));
+      }
+      if (value == null) {
```

Is this even possible with the splitter? In any case, don't think this method needs to check for it. Should be the responsibility of the caller to make sure the map passed in is valid.

---

The Splitter allows the value to be null, if the key is present.
This is used to validate, if a valid (not null) output field name is provided or not, to hold the generated feature vector value.
Please let me know if this is okay or should we throw a runtime exception during StructuredRecord.Builder()